lib/ProductOpener/Redis.pm

```diff
@@ -14,39 +25,96 @@ BEGIN {
 
 use vars @EXPORT_OK;
 
-use Log::Any qw($log);
-use ProductOpener::Config2;
-use Redis::Client;
+use Log::Any qw/$log/;
+use ProductOpener::Config2 qw/$redis_url/;
+use Redis;
+
+=head2 $redis_client
+The connection to redis
+=cut
+my $redis_client;
+
 
-sub init_redis() {
+=head2 init_redis($is_reconnect=0)
 
-	$log->debug("init_redis", {redis_url_env => $ENV{REDIS_URL}, redis_url => $ProductOpener::Config2::redis_url})
+init $redis_client or re-init it if we where disconnected
+
+it is uses  ProductOpener::Config2::redis_url
+
+=head3 Arguments
+
+=head4 bool $is_reconnect
+
+This is informative.
+If true we will display a warning if we have no redis_url.
+
+=cut
+
+sub init_redis($is_reconnect=0) {
+	$log->debug("init_redis", {redis_url => $redis_url})
 	  if $log->is_debug();
 
-	$log->warn("REDIS_URL env", {redis_url => $ENV{REDIS_URL}}) if $log->is_warn();
-	if ($ProductOpener::Config2::redis_url eq "") {
-		$log->warn("Redis URL not provided for search indexing", {error => $@}) if $log->is_warn();
-		return;
+	if (((! defined $redis_url) || ($redis_url eq "")) && ! $is_reconnect) {
+		$log->warn("Redis URL not provided for search indexing") if $log->is_warn();
 	}
-	my $redis_client;
-	eval {$redis_client = Redis::Client->new(host => $ProductOpener::Config2::redis_url);};
+	eval {
+		$redis_client = Redis->new(
+			server => $redis_url,
+			# we don't want to sacrifice too much performance for redis problems
+			cnx_timeout => 1,
+			write_timeout => 1,
+		);
+	};
 	if ($@) {
 		$log->warn("Error connecting to Redis", {error => $@}) if $log->is_warn();
+		$redis_client = -1;  # this ask for eventual reconnection
```

Comment:
It works, but it took me a while to understand that the purpose of the variable $is_reconnect is only to avoid warning multiple times about $redis_url not being set.

I think it would much simpler to declare something like:

my $sent_warning_about_missing_redis_url = 0;

at the beginning after "my $redis_client;", and just test and set this variable once the warning has been sent.

That way we don't have to have a special -1 value for $redis_client: it's either a valid redis client object, or undef.

Reply:
agree.